readme.md

```diff
@@ -70,6 +83,46 @@ example:
 }
 ```
 
+## Upgrading from earlier versions of `winston-graylog2`
+
+Since `winston@3.x` relies heavily on a very powerful set of formatters there are a few formatting
+actions that `winston-graylog2` no longer needs to do. The two primary cases are the inclusion of
+a `meta` object, and converting errors so that the stack trace is included in the log message
+rather than just the name of the error.
+
+`winston@3.x` includes an excellent formatter for dealing with `meta`, conveniently named
+`metadata`. To use it, you can either grab it from the `winston.format` object, or use the one on
+`logform.format`. See [the metadata formatter docs](https://github.com/winstonjs/logform#metadata)
+for more details.
+
+For formatting Errors, `logform` (used under the hood by `winston.format`) also includes an
+excellent formatter which gives you the option to include the stack trace in the logged message.
+**However**, the error formatter is only available with `logform@^2.1.0`, so at the time of this
+writing you must explicitly require it rather than using `winston.format`.
+
+In order to get functionality identical to earlier versions of `winston-graylog2`, use both of
+these formatters together.
+
+```javascript
+var winston = require('winston');
+var { format } require('logform');
+var WinstonGraylog2 = require('winston-graylog2');
+
+var options = { ...<your config options here>... };
+var logger = winston.createLogger({
+  exitOnError: false,
+  formatters: format.combine(
+    format.errorr({ stack: true }),
+    format.metadata(),
+  ),
+  transports: [
+    new WinstonGraylog2(options)  ],
+});
+
+logger.info({ message: 'this is an info message', meta: 42 });
+logger.error({ message: new Error(FakeError), meta: 96 });
```

Comment:
@jeremy-j-ackso awesome! Now it's mostly compatible with the previous version.

One minor note: let's update the readme and add the following lines here:

```
 logger.info('my message', { reason: 'whatever', promise: 'whenever'});
 logger.info(new Error("Oh noes!"));
```

To indicate that you can just pass metadata as a separate object (like the previous version of winston-graylog2 allowed to do so).

Reply:
That should be fine, but I'll definitely double-check with my example project that your suggestion will work.